service-clients-gen/src/main/java/com/flipkart/poseidon/serviceclients/generator/ServiceGenerator.java

```diff
@@ -447,6 +454,12 @@ private void addImplMethods(ServiceIDL serviceIdl, JCodeModel jCodeModel, JDefin
         method._throws(jCodeModel.directClass("UnsupportedOperationException"));
         method.body()._return(JExpr.ref("VERSION"));
 
+        methodReturnType = jCodeModel.ref(ObjectMapper.class);
```

Comment:
This block should be generated only if (serviceIdl.getService().getObjectMapperClass() != null)?